src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -3249,6 +3249,12 @@ internal uint GetRefEscape(BoundExpression expr, uint scopeOfTheContainingExpres
                     }
 
                     return GetRefEscape(assignment.Left, scopeOfTheContainingExpression);
+
+                case BoundKind.Conversion:
+                    Debug.Assert(expr is BoundConversion conversion &&
+                        (!conversion.Conversion.IsUserDefined ||
+                        conversion.Conversion.Method.RefKind == RefKind.None));
```

Comment:
This solution looks good to me. We only have a need to get/check the ref escape of an expression, when it is something we can take a reference to. For expressions which we can't take a reference to, it's reasonable for this method to just return `scopeOfTheContainingExpression`.